Fix *even* more miscellaneous oddities around the class reference

Mickeon opened the pull request:
Continuation of https://github.com/godotengine/godot/pull/99002 and https://github.com/godotengine/godot/pull/95774

This PR addressed yet **yet** another huge chunk of class reference oddities I remembered to note down for later while localizing the class reference. Some of these are clear typos, some are inconsistencies, some are just awkward.

Oh boy this will be nasty for localization.

Of particular note:
- Replaced the single use of "govern" with a more common verb
- Elaborated slightly more on **DisplayServer**'s `screen_get_image` and `screen_get_image_rect`
- ~~Tweaked EditorExportPlatformAndroid's `screen/immersive_mode` to actually mention immersive mode~~
- Went ballistic on Skeleton3D's `metadata`-related descriptions
- Filled many glTF definitions with appropriate `[code]` tags.
- Fixed a common "will be return" typo in the **SCALING_MODE** constants
- Fixed some missing articles (especially where it looked inconsistent with the rest)
- Fixed `vertex_buffer_create` glaringly missing what it actually created.
- Fixed **OS**'s `read_buffer_from_stdin` mentioning the wrong method
- Fixed **GLTFAccessor**'s `accessor_type` mentioning corresponding **int** values even though it's an enum
	- The description was copied from the deprecated `type` property without accounting for this

Mickeon committed:
Fix *even* more miscellaneous oddities around the class reference

syntaxerror247 commented on platform/android/doc_classes/EditorExportPlatformAndroid.xml:
```diff
@@ -597,7 +597,7 @@
 		</member>
 		<member name="screen/immersive_mode" type="bool" setter="" getter="">
-			If [code]true[/code], hides navigation and status bar. See [method DisplayServer.window_set_mode] to toggle it at runtime.
+			If [code]true[/code], enables immersive mode, which hides the navigation and status bar. Set [method DisplayServer.window_set_mode] to change this at runtime.
```
```suggestion
			If [code]true[/code], hides the navigation and status bar. Use [method DisplayServer.window_set_mode] to change this at runtime.
```
Saying "enables immersive mode" is not accurate because, on Android, immersive mode is controlled using a combination of flags rather than a single method that directly enables or disables it.

Mickeon replied:
While I accept the suggestion, it may be a good idea to hint immersive mode is involved. The only way to know is by looking at the export property's name itself, which doesn't... really count for this purpose.

There's also other places in the class reference that refer to certain settings as "enabling immersive mode". Namely:
https://github.com/godotengine/godot/blob/ef1153baf3343470b65258421a163c92455d32b3/doc/classes/Window.xml#L829

syntaxerror247 replied:
In other places, immersive mode points to this export setting. It might be better to link to this export setting.

Mickeon replied:
Well, I _can_ do it in this PR but I don't have to...

AThousandShips commented on doc/classes/CodeEdit.xml:
```diff
@@ -588,7 +588,7 @@
 		</member>
 		<member name="symbol_tooltip_on_hover" type="bool" setter="set_symbol_tooltip_on_hover_enabled" getter="is_symbol_tooltip_on_hover_enabled" default="false">
-			Set when a word is hovered, the [signal symbol_hovered] should be emitted.
+			If [code]true[/code], when hovering over a word the [signal symbol_hovered] signal is emitted.
```
```suggestion
			If [code]true[/code], the [signal symbol_hovered] signal is emitted when hovering over a word.
```

AThousandShips commented on doc/classes/DisplayServer.xml:
```diff
@@ -1908,7 +1908,7 @@
 		</constant>
 		<constant name="FEATURE_HIDPI" value="12" enum="Feature">
-			Display server supports querying the operating system's display scale factor. This allows for [i]reliable[/i] automatic hiDPI display detection, as opposed to guessing based on the screen resolution and reported display DPI (which can be unreliable due to broken monitor EDID). [b]Windows, Linux (Wayland), macOS[/b]
+			Display server supports querying the operating system's display scale factor. This allows to automatically detect the hiDPI display [i]reliably[/i], instead of guessing based on the screen resolution and the display's reported DPI (which may be unreliable due to broken monitor EDID). [b]Windows, Linux (Wayland), macOS[/b]
```
```suggestion
			Display server supports querying the operating system's display scale factor. This allows automatically detecting the hiDPI display [i]reliably[/i], instead of guessing based on the screen resolution and the display's reported DPI (which may be unreliable due to broken monitor EDID). [b]Windows, Linux (Wayland), macOS[/b]
```

AThousandShips commented on doc/classes/Mesh.xml:
```diff
@@ -359,7 +359,7 @@
 		</constant>
 		<constant name="ARRAY_FLAG_USE_DYNAMIC_UPDATE" value="67108864" enum="ArrayFormat" is_bitfield="true">
-			Flag indices that the mesh data will use [code]GL_DYNAMIC_DRAW[/code] on GLES. Unused on Vulkan.
+			Flag indicates that the mesh data will use [code]GL_DYNAMIC_DRAW[/code] on GLES. Unused on Vulkan.
```
```suggestion
			Flag used to mark that the mesh data will use [code]GL_DYNAMIC_DRAW[/code] on GLES. Unused on Vulkan.
```
Matching the nearby flags

AThousandShips commented on doc/classes/RenderingServer.xml:
```diff
@@ -4679,7 +4679,7 @@
 		</constant>
 		<constant name="ARRAY_FLAG_USE_DYNAMIC_UPDATE" value="67108864" enum="ArrayFormat" is_bitfield="true">
-			Flag indices that the mesh data will use [code]GL_DYNAMIC_DRAW[/code] on GLES. Unused on Vulkan.
+			Flag indicates that the mesh data will use [code]GL_DYNAMIC_DRAW[/code] on GLES. Unused on Vulkan.
```
```suggestion
			Flag used to mark that the mesh data will use [code]GL_DYNAMIC_DRAW[/code] on GLES. Unused on Vulkan.
```

AThousandShips commented on doc/classes/SkeletonModifier3D.xml:
This file also has a "there is [AnimationMixer]" which should be "there is an [AnimationMixer]"

Mickeon replied:
Ooooh, there's... many things like that on the animation side...

AThousandShips commented on modules/gltf/doc_classes/GLTFState.xml:
```diff
@@ -25,7 +25,7 @@
 			<param index="1" name="deduplication" type="bool" />
 			<description>
-				Appends the given byte array data to the buffers and creates a [GLTFBufferView] for it. The index of the destination [GLTFBufferView] is returned. If [param deduplication] is [code]true[/code], the buffers will first be searched for duplicate data, otherwise new bytes will always be appended.
+				Appends the given byte array [param data] to the buffers and creates a [GLTFBufferView] for it. The index of the destination [GLTFBufferView] is returned. If [param deduplication] is [code]true[/code], the buffers is first searched for duplicate data, otherwise new bytes are always appended.
```
```suggestion
				Appends the given byte array [param data] to the buffers and creates a [GLTFBufferView] for it. The index of the destination [GLTFBufferView] is returned. If [param deduplication] is [code]true[/code], the buffers are first searched for duplicate data, otherwise new bytes are always appended.
```

bruvzg commented on doc/classes/DisplayServer.xml:
```diff
@@ -1088,18 +1088,18 @@
+				Returns a screenshot of the [param screen].
 				[b]Note:[/b] This method is implemented on Linux (X11), macOS, and Windows.
-				[b]Note:[/b] On macOS, this method requires "Screen Recording" permission, if permission is not granted it will return desktop wallpaper color.
+				[b]Note:[/b] On macOS, this method requires the "Screen Recording" permission. If permission is not granted, this method returns an [Image] filled with the desktop wallpaper's color.
```
> desktop wallpaper color.

Actually it's an image app own windows on top of desktop wallpaper (no icons/widgets, or other app windows aren't visible), not filled with color.

bruvzg replied:
Sorry, missed this, here's an example of what app capture vs. real screen contents:

Real screen:

<img width="1504" alt="Screenshot 2025-04-04 at 00 11 35" src="https://github.com/user-attachments/assets/6630fa85-322d-4c56-b1be-746cea9343b5" />

What's app without permission capture:
<img width="1504" alt="a" src="https://github.com/user-attachments/assets/5435ad4c-0843-4cd3-8d37-bbc8254df1a3" />

Mickeon replied:
Alright, thank you. I have updated the note to be hopefully more accurate.

aaronfranke reviewed (approved):
Read through the whole thing, looks good to me, except for the conflict.

Mickeon commented:
Rebased after minor conflicts in `OS.xml`.
As @bruvzg has not elaborated further in https://github.com/godotengine/godot/pull/104333#discussion_r2005918688 I have reverted part of the changes to those descriptions in DisplayServer, for now. They can be done separately for more concentrated feedback.

AThousandShips commented on doc/classes/DisplayServer.xml:
```diff
@@ -1088,18 +1088,18 @@
+				Returns a screenshot of the [param screen].
 				[b]Note:[/b] This method is implemented on Linux (X11), macOS, and Windows.
-				[b]Note:[/b] On macOS, this method requires "Screen Recording" permission, if permission is not granted it will return desktop wallpaper color.
+				[b]Note:[/b] On macOS, this method requires the "Screen Recording" permission. If permission is not granted, this method returns a screenshot that may only contain the desktop wallpaper, the current application's window, and other related UI elements.
```
If "these will be returned, without exception" is meant
```suggestion
				[b]Note:[/b] On macOS, this method requires the "Screen Recording" permission. If permission is not granted, this method returns a screenshot that will only contain the desktop wallpaper, the current application's window, and other related UI elements.
```
Or, if "it is possible for these to be returned":
```suggestion
				[b]Note:[/b] On macOS, this method requires the "Screen Recording" permission. If permission is not granted, this method returns a screenshot that might only contain the desktop wallpaper, the current application's window, and other related UI elements.
```

Or some similar construction, "may" conventionally means "to be permitted to" and not "to be possible to", and this case is easily confusing because of that ambiguity

Mickeon replied:
It's absolutely not a guarantee and it depends on circumstance. I read your thoughts on using "may" or not but you should further discuss this.

bruvzg replied:
No permission, can't see other apps, there are no other conditions.

AThousandShips replied:
This is based on what was done on the other PR

But "will" would be appropriate here, "might" would mean "it can show these, or others"

AThousandShips replied:
But regardless "may" is not correct here as it's not a matter of the screenshot itself only being permitted to have these things

Mickeon replied:
What I do mean is that it's not guaranteed to be all three. So, perhaps it, actually, it should be "**MIGHT** only contain the desktop wallpaper, the current application's window, **OR** other related UI elements."

AThousandShips replied:
Not with "only", then it instead means "it might contain this, or other things", if it said "it will contain" it would be a question of "it will contain exactly these"

bruvzg replied:
Maybe it should be the opposite: "will not include other application windows and os elements not related to the application". This note exists to show the limitation, not to list all possible elements of the screenshot.

AThousandShips replied:
That sounds far better actually

Mickeon replied:
I love bikeshedding so much actually

AThousandShips commented on doc/classes/DisplayServer.xml:
```diff
@@ -1088,18 +1088,18 @@
+				Returns a screenshot of the screen region defined by [param rect].
 				[b]Note:[/b] This method is implemented on macOS and Windows.
-				[b]Note:[/b] On macOS, this method requires "Screen Recording" permission, if permission is not granted it will return desktop wallpaper color.
+				[b]Note:[/b] On macOS, this method requires the "Screen Recording" permission. If permission is not granted, this method returns a screenshot that may only contain the desktop wallpaper, the current application's window, and other related UI elements.
```
Same here

AThousandShips commented on doc/classes/DisplayServer.xml:
```diff
@@ -1899,7 +1899,7 @@
 		</constant>
 		<constant name="FEATURE_HIDPI" value="12" enum="Feature">
-			Display server supports querying the operating system's display scale factor. This allows for [i]reliable[/i] automatic hiDPI display detection, as opposed to guessing based on the screen resolution and reported display DPI (which can be unreliable due to broken monitor EDID). [b]Windows, Linux (Wayland), macOS[/b]
+			Display server supports querying the operating system's display scale factor. This allows automatically detecting the hiDPI display [i]reliably[/i], instead of guessing based on the screen resolution and the display's reported DPI (which may be unreliable due to broken monitor EDID). [b]Windows, Linux (Wayland), macOS[/b]
```
```suggestion
			Display server supports querying the operating system's display scale factor. This allows automatically detecting the hiDPI display [i]reliably[/i], instead of guessing based on the screen resolution and the display's reported DPI (which might be unreliable due to broken monitor EDID). [b]Windows, Linux (Wayland), macOS[/b]
```

AThousandShips commented on platform/ios/doc_classes/EditorExportPlatformIOS.xml:
```diff
@@ -106,10 +106,10 @@
 		</member>
 		<member name="entitlements/increased_memory_limit" type="bool" setter="" getter="">
-			Enable if app may perform better with a higher memory limit. [url=https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_kernel_increased-memory-limit]com.apple.developer.kernel.increased-memory-limit[/url].
+			If [code]true[/code], hints that the app may perform better with a higher memory limit. See [url=https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_kernel_increased-memory-limit]com.apple.developer.kernel.increased-memory-limit[/url].
```
```suggestion
			If [code]true[/code], hints that the app might perform better with a higher memory limit. See [url=https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_kernel_increased-memory-limit]com.apple.developer.kernel.increased-memory-limit[/url].
```

AThousandShips commented on doc/classes/DisplayServer.xml:
```diff
@@ -1088,18 +1088,18 @@
+				Returns a screenshot of the [param screen].
 				[b]Note:[/b] This method is implemented on Linux (X11), macOS, and Windows.
-				[b]Note:[/b] On macOS, this method requires "Screen Recording" permission, if permission is not granted it will return desktop wallpaper color.
+				[b]Note:[/b] On macOS, this method requires the "Screen Recording" permission. If permission is not granted, this method returns a screenshot that will not include other application windows and OS elements not related to the application.
```
```suggestion
				[b]Note:[/b] On macOS, this method requires the "Screen Recording" permission. If permission is not granted, this method returns a screenshot that will not include other application windows or OS elements not related to the application.
```
Final detail, think "I don't have a dog or a cat", in this case this means "it won't include A and B together, but can include A, or B, or neither"

AThousandShips commented on doc/classes/DisplayServer.xml:
```diff
@@ -1088,18 +1088,18 @@
+				Returns a screenshot of the screen region defined by [param rect].
 				[b]Note:[/b] This method is implemented on macOS and Windows.
-				[b]Note:[/b] On macOS, this method requires "Screen Recording" permission, if permission is not granted it will return desktop wallpaper color.
+				[b]Note:[/b] On macOS, this method requires the "Screen Recording" permission. If permission is not granted, this method returns a screenshot that will not include other application windows and OS elements not related to the application.
```
```suggestion
				[b]Note:[/b] On macOS, this method requires the "Screen Recording" permission. If permission is not granted, this method returns a screenshot that will not include other application windows or OS elements not related to the application.
```

Repiteo commented:
Thanks!
